src/main/java/io/nats/client/SubscribeOptions.java

```diff
@@ -105,60 +66,35 @@ public long getPullBatchSize() {
         return pull;
     }
 
-
-    private static void checkStreamName(String stream) {
-        if (stream == null || stream.length() == 0 || stream.contains(">") ||
-        stream.contains(".") || stream.contains("*")) {
-            throw new IllegalArgumentException("stream cannot be null, empty, tokenized, or wildcarded");
-        }
-    }
     /**
-     * Setting this enables pull based consumers.  This sets the batch size the server
-     * will send.  Default is 0 (disabled).
-     * @param batchSize number of messages to request in poll. 
+     * Get a default instance of SubscribeOptions
+     * @return the instance
      */
-    public void setPull(long batchSize) {
-        if (batchSize <= 0) {
-            throw new IllegalArgumentException("Batch size must be greater than zero");
-        }
-        pull = batchSize;
+    public static SubscribeOptions getInstance() {
+        return new Builder().build();
     }
 
     /**
-     * Setting this enables pull based consumers.  This sets the batch size the server
-     * will send.  Default is 0 (disabled).
-     * @param stream the name of the stream
-     * @param consumer the name of the consumer
-     * @param batchSize number of messages to request in poll. 
-     */
-    public void setPullDirect(String stream, String consumer, long batchSize) {
-        checkStreamName(stream);
-        setPull(batchSize);
-        this.stream = stream;
-        this.consumer = consumer;
-    }    
-
-    /**
-     * Attaches to a consumer. 
-     * @param stream the name of the stream the consumer is attached to.
-     * @param consumer the consumer name.
+     * Creates a builder for the publish options.
+     * @return the builder.
      */
-    public void attach(String stream, String consumer) {
-        checkStreamName(stream);
-        if (consumer == null) {
-            throw new IllegalArgumentException("Consumer cannot be null");
+    public static SubscribeOptions getInstance(SubscribeOptions options) {

```

Comment:
Perhaps this should be private?  I'm not sure what value this provides the user since the options are immutable.

Reply:
This is used in NatsJetStream.createSubscription which is in the impl package, SubscribeOptions is in client so I can't make it package.

Reply:
Ah, gotcha.